### PR TITLE
fetch physical and logical block size if available

### PR DESCRIPTION
**Pull request by richardstephens**
We need to know the udev-reported logical block size

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -871,17 +879,22 @@ fn test_get_device_info() {
...
+            let val = val_str.to_str().unwrap_or("0").parse::<u64>().unwrap_or(0);
+            Some(val)
         }
         None => None,
     }
 }
 
+#[cfg(target_os = "linux")]
+fn get_size(device: &udev::Device) -> Option<u64> {
+    // 512 is the block size
```
Comment (cholcombe973) [inline src/lib.rs:894]: I think this isn't always true. There's some 4k block size devices out there. It's probably worth taking a parameter for this or querying the size from the device at runtime. What do you think?

Comment (richardstephens) [reply src/lib.rs:894]: I'm still digging into this, but I have a test machine with an NVMe disk formatted to 4k sectors. On this machine, the size presented from udev is still a multiple of 512 bytes.

Comment (cholcombe973) [reply src/lib.rs:894]: That's interesting. Alright let's land this. Do you have other PR's planned?

Comment (richardstephens) [reply src/lib.rs:894]:
```
root@debian-0cc70cf1:~# smartctl -a /dev/nvme1 |grep "Namespace 1"
Namespace 1 Size/Capacity:          1,000,204,886,016 [1.00 TB]
Namespace 1 Formatted LBA Size:     4096
Namespace 1 IEEE EUI-64:            6479a7 4db00002a0
root@debian-0cc70cf1:~# udevadm info -a /dev/nvme1n1|grep size
    ATTR{integrity/tag_size}=="0"
    ATTR{queue/hw_sector_size}=="4096"
    ATTR{queue/logical_block_size}=="4096"
    ATTR{queue/max_segment_size}=="4294967295"
    ATTR{queue/minimum_io_size}=="4096"
    ATTR{queue/optimal_io_size}=="0"
    ATTR{queue/physical_block_size}=="4096"
    ATTR{size}=="1953525168"
    ATTRS{sqsize}=="1023"
root@debian-0cc70cf1:~# 
```
```
>>> 1953525168 *512
1000204886016
```

Comment (richardstephens) [reply src/lib.rs:894]: I googled around a bit but didn't find any documentation. I'll dig into the udev source to see if i can find anything more.

Comment (richardstephens) [reply src/lib.rs:894]: Found [this](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/types.h?h=v6.3-rc5#n120). I'm not sure how safe an assumption it is that this maps up through udev.

> That's interesting. Alright let's land this. Do you have other PR's planned?

Well, I still want to fix `list_nvme_controllers()` when I have a free moment. But I think we have all the functionality we need for now.

**Pull request merged**